proposals/0000-cases.md

```diff
@@ -0,0 +1,676 @@
+      it could be used instead of `-XLambdaCase` and would be able to match on
+      multiple patterns.
+ 2. The extension `-XMultiWayIf` essentially introduces standalone guards,
+    simplifying the use of guards that aren't at the outermost level of a
+    function declaration or case-expression. Among other things, this made it
+    easier to use guards inside of lambda expressions.
+    - If there were an expression similar to lambda expressions that could have
+      guards and wasn't required to have at
+      least one parameter, it could be used instead of `-XMultiWayIf`. This
+      includes all uses of `-XMultiWayIf`, not just those inside of lambdas
+      (see Example section).
+ 3. During the implementation of `-XLambdaCase`,
+    [some suggested](https://gitlab.haskell.org/ghc/ghc/issues/4359#note_51110)
+    allowing lambda expressions to have multiple clauses. This was not
+    implemented: The most obvious approach of turning `\` into a layout herald
+    had the disadvantage of making some common idioms invalid.
+    - This could be circumvented by introducing a new expression that isn't required
+      to be backwards compatible with existing idioms.
+
+## Proposed Change Specification
+
+#### BNF Changes
+
+**Bold** indicates changes to the existing BNF.
+
+<table>
+    <tr>
+        <td><i>lexp</i></td><td>&rarr;</td><td>&hellip;</td>
+    </tr>
+    <tr>
+        <td></td><td>|</td><td><tt>\case</tt> <tt>{</tt> <i>alts</i> <tt>}</tt></td><td>(<tt>\case</tt> expression)</td>
+    </tr>
+    <tr>
+        <td></td><td><b>|</b></td><td><b><tt>\cases</tt> <tt>{</tt> <i>nalts</i> <tt>}</tt></b></td><td><b>(<tt>\cases</tt> expression)</b></td>
+    </tr>
+    <tr>
+        <td><i>alts</i></td><td>&rarr;</td><td><i>alt</i><sub>1</sub> <tt>;</tt> &hellip; <tt>;</tt> <i>alt</i><sub>m</sub></td><td>(m &ge; 1)</td>
+    </tr>
+    <tr>
+        <td><i>alt</i></td><td>&rarr;</td><td><i>pat</i> <tt>-></tt> <i>exp</i> [ <tt> where </tt> <i>decls</i> ]</td>
+    </tr>
+    <tr>
+        <td></td><td>|</td><td><i>pat</i> <i>gdpat</i> [ <tt> where </tt> <i>decls</i> ]</td>
+    </tr>
+    <tr>
+        <td></td><td>|</td><td></td><td>(empty alternative)</td>
+    </tr>
+    <tr>
+        <td><b><i>nalts</i></b></td><td><b>&rarr;</b></td><td><b><i>nalt</i><sub>1</sub> <tt>;</tt> &hellip; <tt>;</tt> <i>nalt</i><sub>m</sub></b></td><td><b>(m &ge; 1)</b></td>
+    </tr>
+    <tr>
+        <td><b><i>nalt</i></b></td><td><b>&rarr;</b></td><td><b>[ <i>apat</i><sub>1</sub> &hellip; <i>apat</i><sub>n</sub> ] <tt>-></tt> <i>exp</i> [ <tt> where </tt> <i>decls</i> ]</b></td><td><b>(n &ge; 0)</b></td>
+    </tr>
+    <tr>
+        <td></td><td><b>|</b></td><td><b>[ <i>apat</i><sub>1</sub> &hellip; <i>apat</i><sub>n</sub> ] <i>gdpat</i> [ <tt> where </tt> <i>decls</i> ]</b></td><td><b>(n &ge; 0)</b></td>
+    </tr>
+    <tr>
+        <td></td><td><b>|</b></td><td></td><td><b>(empty alternative)</b></td>
+    </tr>
+<table>
```

Comment:
Hang on... for \case we can have zero alts, so (m >= 0) in the `alts` equation, surely?

Reply:
You're right, Ctrl-C Ctrl-V artifact